docs/internal/generics.md

```diff
@@ -156,3 +156,110 @@ let z : U32 & imm = obj.convert(y); // ERROR: Not able to define what type y is
 
 ## Specialisation
 
+Type parameters make classes generic to share implementation for a range of different types ([parametric polymorphism](https://en.wikipedia.org/wiki/Parametric_polymorphism)).
+But concrete classes need monomorphic types to be defined at compile time, so that we can have a concrete implementation of a parametric type.
+This conversion is called [reification](https://en.wikipedia.org/wiki/Reification_(computer_science)) and is done via a reachability analysis of all possible definitions of each polymorphic type.
+
+Example:
+```ts
+class Generic[A] { field : A & imm; ... };
+...
+// Here, `A` is `U32`
+let foo : Generic[U32] = ...;
+
+// Here, `A` is `MyClass`
+let bar : Generic[MyClass] = ...;
+```
+
+In the example above, there is one polymorphic class, `Generic[A]` where `A` is the polymorphic type parameter, and two concrete classes (specialisations): `Generic[U32]` and `Generic[MyClass]`.
+
+The compiler generates the required specialisations and replaces each definition with its respective concrete classes, then removes the generic class.
+Since generic classes can only be used by specialisation, all other (potentially infinite) possibilities of types parameters don't exist in the program, so the generic class is elided and we're left with only the specialisations.
```

Comment:
Note that this is not always going to be guaranteed by the language.  

Once we add support for loadable code, generics that are exposed across a program / plugin boundary will not be fully reified and will end up doing some dynamic dispatch.  

Before then, the compiler is always free to merge similar instantiations of generics and add some dynamic dispatch back.

Reply:
In my view, such a generic interface will be in a monomorphic (union) type with a `match` statement, so "fully reified".

Reply:
This is not possible with separate compilation.

Reply:
I thought we wouldn't have separate compilation in Verona.

Reply:
Ok, I mention loadable code here so that there are no doubts. But loadable code is still not defined, so I want to keep the discussion of eliding the whole thing if static.